src/openforms/registrations/contrib/objects_api/config.py

```diff
@@ -92,3 +136,41 @@ class ObjectsAPIOptionsSerializer(JsonSchemaSerializerMixin, serializers.Seriali
         ],
         required=False,
     )
+
+    # V2 only fields:
+    variables_mapping = ObjecttypeVariableMappingSerializer(
+        label=_("variables mapping"),
+        many=True,
+        required=False,
+    )
+
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        v1_only_fields = {"content_json", "payment_status_update_json"}
+        v2_only_fields = {"variables_mapping"}
+
+        version = attrs["version"]
+
+        if version == 1:
+            v1_forbidden_fields = v2_only_fields.intersection(attrs)
+            if v1_forbidden_fields:
+                raise ValidationError(
+                    {
+                        k: _(
+                            "{field_name} shouldn't be provided when version is 1"
+                        ).format(field_name=k)
+                        for k in v1_forbidden_fields
+                    }
+                )
+        if version == 2:
+            v2_forbidden_fields = v1_only_fields.intersection(attrs)
+            if v2_forbidden_fields:
+                raise ValidationError(
+                    {
+                        k: _(
+                            "{field_name} shouldn't be provided when version is 2"
+                        ).format(field_name=k)
+                        for k in v2_forbidden_fields
+                    }
+                )
```

Comment:
since those are distinct cases, I think pattern matching could be leveraged here for exhaustiveness:

```py
match version:
    case 1:
        ...
    case 2:
        ...
    case _:
        raise ValueError(f"Unknown version {version}")
```